### ISSUE TITLE
A version for modded EEprom radios?

### ISSUE THREAD

**Issue by SutliffeProductions** [armel/uv-k5-firmware-custom#144]
Hi. 
This is by far the most feature rich firmware I've seen so far really well done. Is there an possibility down the line, for a version which includes as many features as possible (pong, SMS, and custom boot images would be great) designed for radios with a larger eeprom mod installed? as it's a fairly easy mod to do, and would mean you are not as storange limited.

Keep up the good work!

**Comment by prokrypt** [comment]
We will still be limited by 64KB flash, and it will take quite some work to get static resources moved to eeprom... Hardware mods are easy. The software part, not so much.

**Comment by SutliffeProductions** [comment]
> We will still be limited by 64KB flash, and it will take quite some work to get static resources moved to eeprom... Hardware mods are easy. The software part, not so much.

I'm not sure what you mean, I could be wrong, but I was under the impression that the 64kb was the program EEprom space, as that is the only storage onboard. Apart from 8k of ram?

**Comment by armel** [comment]
Hello @SutliffeProductions and @prokrypt :)

@prokrypt is right.

@SutliffeProductions, your Quansheng K5 embed a N32G030. It's  a 32 bits microcontroller (based on Arm® Cortex®-M0) with only 64 KB flash memory. 

As I've explained many times, don't confuse the EEPROM (which stores calibration data, settings and memory channels) with the microcontroller's flash memory (which stores the firmware) 😌 These are 2 different things. EEPROM is not (never) use to store or execute firmware. By the way, this EEPROM is very slow. 

In short, even if you update your EEPROM, you are still limited to 64 KB of flash memory, embedded in the microcontroller, to store your firmware.

73'

Armel F4HWN.

**Comment by svh1985** [comment]
Maybe now it’s a bit more relevant, see the below posts .The modder even used @armel’s firmware. 😆

[https://makeprogress.ee/blog/from-feature-tetris-to-full-power-the-uv-k5-flex-pcb-brain-hack](url)

[https://www.reddit.com/r/amateurradio/s/FYChvDOy1C](url)

**Comment by armel** [comment]
@svh1985 This project is very interesting. However, as with all other hardware mods, only a small minority of users will actually give it a try. It therefore becomes a very specific development, similar for example to the CEC firmware designed for the SI4732 mod.

So as for me, even though I’ll be following this project, I won’t be doing anything beyond that. But since my firmware is Open Source, anyone can fork it and dedicate it to this mod.
